pkgs/development/idris-modules/build-idris-package.nix

```diff
@@ -2,26 +2,27 @@
 #
 # args: Additional arguments to pass to mkDerivation. Generally should include at least
 #       name and src.
-{ stdenv, idris, gmp }: args: stdenv.mkDerivation ({
-  preHook = ''
```

Comment:
have you tried `postHook`, that runs after the default setup.sh but before `genericBuild`

Reply:
Please forgive my ignorance but now I'm curious - how would this part https://github.com/NixOS/nixpkgs/pull/33838/files#diff-852bb06e74b1ca7f9271dc4dd42aaca9R20 be implemented when using `postHook`? (and sorry for the noise - I'm struggling to figure out how the various pieces fit together here).

Reply:
I only looked through this very briefly so I might be missing something.

The preHook runs at the very start of `$stdenv/setup` only a few things like helpers and PATH are defined at this point. On the other had the postHook will run at the very end of `$stdenv/setup` when all of the phases, etc. are defined. This is still before all of the phases run but at this point `addEnvHooks` will be available, I think that will work without having to use a setup hook.

Reply:
@LnL7 thank you! It seems to me that `addEnvHooks` would be available in `postHook` but at the point `postHook` adds hooks via  `addEnvHooks` said hooks would have been executed already?

What I'm trying to say is that adding a hook via `addEnvHooks` in `postHook` seems to be to late for it to execute.

(apologies for derailing the discussion, I will shut up now 😆)

Reply:
Oh, you're right, you can ignore the comment then. 😄 The [env hooks](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh#L591) run before that so we need to either move that down or define addEnvHooks earlier.